Features/ManageabilityPkg/Universal/IpmiSmbiosTransferDxe/IpmiSmbiosTransferDxe.c

```diff
@@ -0,0 +1,293 @@
+    Status = gRT->GetVariable (
+                    SMBIOS_HASH_VARIABLE,
+                    &gManageabilityVariableGuid,
+                    NULL,
+                    &StoredHashValueSize,
+                    (VOID *)StoredHashValue
+                    );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_ERROR, "%a: Failed to get UEFI Variable SmbiosHash %r\n", __func__, Status));
+    } else {
+      // Compare StoredHash with currently ComputedHash
+      if (StoredHashValueSize != SHA256_DIGEST_SIZE) {
+        DEBUG ((DEBUG_ERROR, "%a: Invalid Hash Size %u\n", __func__, StoredHashValueSize));
+      }
+
+      if (CompareMem (StoredHashValue, ComputedHashValue, SHA256_DIGEST_SIZE) == 0) {
+        DEBUG ((DEBUG_INFO, "%a: Same Keys , Hash values match\n", __func__));
+        UpdateRequired = FALSE;
+      }
+    }
+  }
+
+  if (UpdateRequired == FALSE) {
+    return UpdateRequired;
+  }
+
+  if (DeleteVar == FALSE) {
+    // ComputedHashValue is valid
+    // store the computed hash and transfer smbios tables if smbios hash variable not found or hash mismatched.
+    Status = gRT->SetVariable (
+                    SMBIOS_HASH_VARIABLE,
+                    &gManageabilityVariableGuid,
+                    EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS,
+                    SHA256_DIGEST_SIZE,
+                    (VOID *)ComputedHashValue
+                    );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_ERROR, "%a: Failed to set UEFI Variable SmbiosHash %r\n", __func__, Status));
+    }
+  }
+
+  return UpdateRequired;
+}
+
+/**
+  This function will send all installed SMBIOS tables to the BMC
+
+  @param  Event    The event of notify protocol.
+  @param  Context  Notify event context.
+**/
+VOID
+EFIAPI
+IpmiSmbiosTransferSendTables (
+  IN EFI_EVENT  Event,
+  IN VOID       *Context
+  )
+{
+  EFI_STATUS                         Status;
+  SMBIOS_TABLE_3_0_ENTRY_POINT       *Smbios30Table;
+  SMBIOS_TABLE_3_0_ENTRY_POINT       *Smbios30TableModified;
+  EDKII_IPMI_BLOB_TRANSFER_PROTOCOL  *IpmiBlobTransfer;
+  UINT16                             Index;
+  UINT16                             SessionId;
+  UINT8                              *SendData;
+  UINT32                             SendDataSize;
+  UINT32                             RemainingDataSize;
+  BOOLEAN                            SmbiosTransferRequired;
+
+  gBS->CloseEvent (Event);
+
+  Status = gBS->LocateProtocol (&gEdkiiIpmiBlobTransferProtocolGuid, NULL, (VOID **)&IpmiBlobTransfer);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: No IpmiBlobTransferProtocol available. Exiting\n", __func__));
+    goto ErrorExit;
+  }
+
+  SmbiosTransferRequired = TRUE;
+  Smbios30Table          = NULL;
+  Status                 = EfiGetSystemConfigurationTable (&gEfiSmbios3TableGuid, (VOID **)&Smbios30Table);
+  if (EFI_ERROR (Status) || (Smbios30Table == NULL)) {
+    DEBUG ((DEBUG_ERROR, "%a: No SMBIOS Table found: %r\n", __func__, Status));

```

Comment:
Do we support 32-bit SMBIOS table?

Reply:
In NVIDIA design, BMC only needs SMBISO 3.0. We can have an enhancement to support 32-bit SMBIOS table later.